docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,705 @@
+)
+
+/* DID document */
+{
+    "context": [
+        "https://www.w3.org/ns/did/v1"
+    ],
+    "id": "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb",
+    "controller": [
+        "did:key:cosmos195rlq2hjnn2tmagskys4xtsnsey6gjljg8zxrn" 
+    ],
+    "verificationMethods" : [
+        // <-- original verification method removed ==> keys have been rotated
+        { 
+            "id": "did:key:cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2#cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2",
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:key:cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2",
+            "blockchainAccountID": "cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2"
+        }
+    ],
+    "authentication": [
+        "did:key:cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2#cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2"  
+    ],
+    "services": []  // <-- the service list is now empty
+}
+
+/* DID metadata */
+{
+  "created": "2021-03-23T06:35:22Z",
+  "updated": "2021-08-23T06:35:22Z",  // <--  update field modified
+  "versionId": "35ba49b94b20ac0a1805aa5283035ea71b003f56e3aeda6a4e9027779fe4aef5", /* hex(blake2b(tx)) */ // <-- new hash computed
+  "deactivated": false
+}
+```
+
+###### Deactivate
+
+A DID can be deactivated using the gRPC message:
+
+```
+MsgDeactivateDid(did string, signerAccount string)
+```
+
+The DID identified by the parameter `did` MUST exist on-chain and the `signerAccount` MUST exist as a verification method (property `blockchainAccountID`) in a verification relationship of type `authentication` or being listed as a DID controller.
+
+This operation MUST remove all the verification methods and controllers and set the metadata property `deactivated` to true. This operation is not reversible.
+
+Example:
+
+```javascript
+/* gRPC message */
+MsgAddDeactivateDid(
+    "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb", 
+    "cash1n90e4s33ljxn00lhucvnmnfjng773efma38dc2"
+)
+
+/* DID document */
+{
+    "context": [
+        "https://www.w3.org/ns/did/v1"
+    ],
+    "id": "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb",
+    "controller": [], // <-- controllers are removed 
+    "verificationMethods" : [] // <-- verification methods are removed
+}
+
+/* DID metadata */
+{
+  "created": "2021-03-23T06:35:22Z",
+  "updated": "2021-09-23T06:35:22Z",  // <--  update field modified
+  "versionId": "e5ca728b93d19daa54180e20eec76d0d5614656c2ee3509df5a21a3abc1249ea", /* hex(blake2b(tx)) */ // <-- new hash computed
+  "deactivated": true  // <-- field deactivated set to true
+}
+```
+
+### Method-specific Properties 
+
+#### DID-core Verification Material
+
+The [Verification Material](https://www.w3.org/TR/did-core/#verification-material) type must be `EcdsaSecp256k1RecoveryMethod2020`.
+
+The content of the verification material can be `blockchainAccountID`, but for interoperability reasons, the verification material should support also `publicKeyHex`. 
```

Comment:
It may be stating the obvious, but would this benefit by stating what the reasons are? I think one reason for illustrative purposes would be ok